File: token.rb
#
# Source Provided by REPL Team : https://gist.github.com/amasad/2304c0f6cf694373fecc
#

require 'openssl'
require 'date'
require 'base64'

def gen_token()
    digest = OpenSSL::Digest.new('sha256')
    secret = 'put your secret here'
    time_created = Time.now.to_i * 1000 # convert to ms
    puts time_created
    hmac = OpenSSL::HMAC.digest(digest, secret, time_created.to_s)
    token = {
        msg_mac: Base64.encode64(hmac).strip,
        time_created: time_created
    }
    puts token
end

gen_token()

File: lib/token.js
// Token Generater file for REPL

var crypto;
var token;

module.exports.generateToken = function () {

	try {
		crypto = require('crypto')
		var secret = require('../config/config').secret

		var timeNow = new Date().getTime()
		var hash = crypto.createHmac('sha256', secret)
									.update(timeNow + '')
									.digest('base64')
		token = {msg_mac : hash, time_created : timeNow}
	} catch(err) {
		console.log('crypto support is disabled')
	}

	return token
}


